### DISCUSSION TITLE
set up a Pacemaker experiment using E3SM

### DISCUSSION THREAD

**Discussion by wenyuz** [E3SM-Project/E3SM#6406] [E3SM model help]
Hi,

I would like set up a Pacemaker experiment using E3SM. A pacemaker experiment restores the regional (e.g. tropical or East Pacific) SST to the model climatology plus historical anomaly. It allows one to study the climate impact of the nudged SST pattern and the contribution of the couple internal mode.

CESM2 has conducted a Pacemaker experiment that restores the Tropical Eastern Pacific SST. Here is their instruction: 
https://www.cesm.ucar.edu/working-groups/climate/simulations/cesm2-pacific-pacemaker
https://www.cesm.ucar.edu/working-groups/climate/simulations/cesm2-pacific-pacemaker/instructions

As I understand the restoring is applied on the ocean grid (POP). 

In MPAS-ocean, we have the "ocean surface restoring" function under the namelist variable "config_use_activetracers_surface_restoring".

In mpas-ocean/src/shared/

mpas_ocn_tendency.F

    ! ocean surface restoring
    if (config_use_tracerGroup_surface_restoring) then
            call mpas_timer_start("surface_restoring_"//groupName)
            call mpas_pool_get_subpool(forcingPool, &
                                   'tracersSurfaceRestoringFields', &
                                    tracersSurfaceRestoringFieldsPool)

            modifiedGroupName = groupName // "PistonVelocity"
            call mpas_pool_get_array(tracersSurfaceRestoringFieldsPool,&
                                     modifiedGroupName, &
                                     tracerGroupPistonVelocity)

            modifiedGroupName = groupName//"SurfaceRestoringValue"
            call mpas_pool_get_array(tracersSurfaceRestoringFieldsPool,&
                                     modifiedGroupName, &
                                     tracerGroupSurfaceRestoringValue)

            ! Note: monthly surface salinity restoring is a special
            ! case for tracer restoring
            call ocn_tracer_surface_restoring_compute(groupName, &
                   nTracersGroup, nCellsAll, tracerGroup,  &
                   tracerGroupPistonVelocity, &
                   tracerGroupSurfaceRestoringValue, &
                   tracerGroupSurfaceFlux, indxSalt, &
                   config_use_surface_salinity_monthly_restoring, &
                   config_salinity_restoring_constant_piston_velocity, &
                   err)
            call mpas_timer_stop("surface_restoring_"//groupName)
         endif ! sfc restoring

and subroutine ocn_tracer_surface_restoring_compute

      !$omp do schedule(runtime) private(iTracer, iLevel)
      do iCell=1,nCells
        iLevel = minLevelCell(iCell)  ! base surface flux restoring on tracer fields in the top layer

        do iTracer=1,nTracers

          ! For monthly salinity restoring, tracersSurfaceRestoringValue contains the zero-mean deltaS
          if (trim(groupName) == 'activeTracers' &
                .and. iTracer == indexSalinity   &
                .and. use_surface_salinity_monthly_restoring) then
          tracersSurfaceFlux(iTracer, iCell) =   tracersSurfaceFlux(iTracer, iCell) + &
                                                 salinity_restoring_constant_piston_velocity * &
                                                 tracersSurfaceRestoringValue(iTracer,iCell)
          else
          tracersSurfaceFlux(iTracer, iCell) =  tracersSurfaceFlux(iTracer, iCell) - &
                                                pistonVelocity(iTracer,iCell) *      &
                                                 (tracers(iTracer, iLevel, iCell) - tracersSurfaceRestoringValue(iTracer,iCell))
          endif
        enddo

I wonder if this function can be applied to restore SST. If so, I wonder what are the detailed procedures to fulfill it.

Alternatively, one may do the SST restoring from the atmospheric grid, just modify the flux from atmosphere to ocean so that the anomalous flux works to restore  SST to observation. But, I am not sure if this will be easier than the ocean way.

Thanks,

Wenyu Zhou
PNNL

**Comment by wenyuz** [comment]
@alicebarthel

**Reply by wenyuz** [reply]
@alicebarthel 

Hi Alice,

I wonder if you know where one can specify the input files (that contain PistonVelocity and SurfaceRestoringValue) and how to construct such files.

Thank you!

Wenyu

**Comment by rljacob** [comment]
E3SM/MPAS-O doesn't have the capability to restore SST over a region while coupling over the rest.  The code mods for POP in step 5 of the Pacemaker instructions haven't been done for MPAS-O.

**Reply by inciente** [reply]
Hello Robert, 

thanks for replying to this thread. 
Just to be completely sure - the pacemaker (or nudging) would simply require adding a prescribed term to the surface forcing routines (CESM2 embeds the extra term within the sensible heat flux). Coupling doesn't need to be deactivated in any regions. 

This type of experiment has been done for EAM (see [this repository](https://github.com/E3SM-Project/E3SM/tree/archive/Sun_JAMES_2019_nudging?tab=readme-ov-file)) and is a crucial tool in climate dynamics research. Are there plans to develop the capability for nudging in MPAS-Ocean? 

Thanks

**Reply by vanroekel** [reply]
The main barrier in implementation of this capability is restoring to a time varying field.  If you wanted to restore to a time invariant field, you can do it now.  But restoring to a time varying field will be difficult.  You would need to follow an approach similar to what we do for sea surface salinity restoring in the code.

The ocean team is in the midst of a complete reconstruction of the code, so this capability is not on the agenda and won't be for at least the next few years.
